### PR TITLE
#11042: Overload complex fw ops

### LINKED ISSUE #11042: Overload complex forward ops for Add and Sub (Set 1 ops)
Tracking Issue : #11041 

List of op to cover : 

- [x] Add
- [x] Sub

### PR DESCRIPTION
**Pull request by Aswinmcw**
Overload forward complex ops with existing api

Tracking Issue : #11042 

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp [hunk 1]
```diff
@@ -217,6 +217,21 @@ Tensor BinaryOperation<binary_op_type, in_place>::operator()(
         input_tensor_a, scalar_tensor_device, dtype, memory_config, optional_output_tensor, activations, input_tensor_a_activation);
 }
 
+template <BinaryOpType binary_op_type, bool in_place>
+ComplexTensor BinaryOperation<binary_op_type, in_place>::operator()(
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp:221]: What are we doing with this template argument here?

Comment (Aswinmcw) [reply ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp:221]: Existing ttnn.add api contains inplace flag as template so we used the same here

Comment (eyonland) [reply ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp:221]: It seems we have a conflicting approach in our api in that an in_place template arg is taking the place of optional_output_tensors.   Leave it as-is on ComplexTensor setup as well for now but we may need to revisit.

**Review by eyonland — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp [hunk 2]
```diff
@@ -217,6 +217,21 @@ Tensor BinaryOperation<binary_op_type, in_place>::operator()(
...
+    const ttnn::MemoryConfig &output_mem_config) {
+    if (binary_op_type == BinaryOpType::ADD) {
+        return ComplexTensor({ ttnn::add(input_a[0], input_b[0], std::nullopt, output_mem_config),
+             ttnn::add(input_a[1], input_b[1], std::nullopt, output_mem_config) });
+    } else if (binary_op_type == BinaryOpType::SUB) {
+        return ComplexTensor({ ttnn::subtract(input_a[0], input_b[0], std::nullopt, output_mem_config),
+             ttnn::subtract(input_a[1], input_b[1], std::nullopt, output_mem_config) });
+    } else {
+        TT_THROW("Unsupported operation");
+    }
```
Comment (patrickroberts) [inline ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp:233]:
```suggestion
    if constexpr (binary_op_type == BinaryOpType::ADD) {
        return ComplexTensor({ ttnn::add(input_a[0], input_b[0], std::nullopt, output_mem_config),
             ttnn::add(input_a[1], input_b[1], std::nullopt, output_mem_config) });
    } else if constexpr (binary_op_type == BinaryOpType::SUB) {
        return ComplexTensor({ ttnn::subtract(input_a[0], input_b[0], std::nullopt, output_mem_config),
             ttnn::subtract(input_a[1], input_b[1], std::nullopt, output_mem_config) });
    } else {
        static_assert(binary_op_type == BinaryOpType::ADD or binary_op_type == BinaryOpType::SUB, "Unsupported operation");
    }
```

**Review by arakhmati — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp (hunk 1 above)
Comment (eyonland) [reply ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp]: @Aswinmcw , take a look at what was recently done by @mouliraj-mcw for the Inplace ops.  This is what I would like done for these versions as well.

Comment (Aswinmcw) [reply ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp]: Yes I will take care of those changes

**Pull request merged**